pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -78,13 +78,14 @@
     private boolean poolMessage;
     
     // Constructor for out-going message
-    public static <T> MessageImpl<T> create(MessageMetadata msgMetadata, ByteBuffer payload, Schema<T> schema) {
```

Comment:
We should not modify this method, maybe it is better to add a new one, and MoP depends on this method.
https://github.com/streamnative/mop/blob/master/mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/PulsarMessageConverter.java#L66

Reply:
`MessageImpl` is implementation class and it should not be used as a contract.